extmod/machine_i2c.c

```diff
@@ -384,51 +379,53 @@ STATIC mp_obj_t machine_i2c_readfrom_mem(size_t n_args, const mp_obj_t *pos_args
     vstr_init_len(&vstr, args[ARG_n].u_int);
```

Comment:
This would need to become `mp_obj_get_int(args[ARG_n].u_obj)` if a common arg array is used.